app/controllers/maintenances_controller.rb

```diff
@@ -1,2 +1,69 @@
 class MaintenancesController < ApplicationController
+  before_action :set_maintenance, only: %i[show edit update destroy]
+  before_action :set_vehicle, only: %i[index create edit update]
```

Comment:
Allows to use @vehicle in destroy for redirection
```suggestion
  before_action :set_vehicle, only: %i[index create edit update destroy]
```